src/Reflection/ClassInspector.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bigcommerce\Injector\Reflection;
+
+use Bigcommerce\Injector\Cache\ServiceCacheInterface;
+use ReflectionClass;
+use ReflectionException;
+
+class ClassInspector
+{
+    public function __construct(
+        private readonly ReflectionClassCache $reflectionClassCache,
+        private readonly ParameterInspector $parameterInspector,
+        private readonly ServiceCacheInterface $serviceCache,
+        private readonly ClassInspectorStats $stats
+    ) {
+    }
+
+    /**
+     * @param string $class
+     * @param string $method
+     * @return void
+     * @throws ReflectionException
+     */
+    public function inspectMethod(string $class, string $method): void
+    {
+        if ($this->classHasMethod($class, $method)) {
+            if ($this->methodIsPublic($class, $method)) {
+                $this->getMethodSignature($class, $method);
+            }
+        }
+    }
```

Comment:
Why does this method need to be public? And even better question is: why do we need it in the first place?

Reply:
This requires some context from the draft pull request to BC app. See my comment [below](https://github.com/bigcommerce/injector/pull/25#issuecomment-2175185418) as well.

https://github.com/bigcommerce/bigcommerce/pull/58084/files#diff-
a8c4e528612ed5da86a4338940e224bec2b149c790d65ca44a35b65d647be9e8R81

Reply:
Oh, I see, makes more sense. I don't like the fact that this method takes in data and returns nothing, mutating the its state.

I am not going to make this a hard request for change, but more of a food for thought: what if we change how this works a bit, i.e. instead of doing orchestration for reflection and caching all in a single class, we can have a class purely doing all the reflection work, and then use a composition and create a "caching wrapper" with fall-through?

Then the caching layer could have a proper warm-up method a bit better aligned with its responsibilities?

Reply:
Good suggestion. I have moved that method to `CachingClassInspector::warmCache()`.